File: POH7/Mizugi/Main.java
// Try POH
// author: Leonardone @ NEETSDKASU
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        Scanner in = new Scanner(System.in);
        
        final long MOD = 1_000_000_000L;
        long n = in.nextLong();
        long r = 1L, b = 0L;
        
        for (long i = 1; i <= n; i++) {
            long t = i;
            while (t % 2L == 0L) {
                t /= 2L;
                b++;
            }
            while (t % 5L == 0L) {
                t /= 5L;
                b--;
            }
            r = (r * t) % MOD;
        }
        
        for (long i = 0;+ i < b; i++) {
            r = (r * 2L) % MOD;
        }
        
        System.out.println(r);
    }
}

File: POH7/Mizugi/Main.vb
' Try POH
' author: Leonardone @ NEETSDKASU
Imports System

Public Class Main
    Public Shared Sub Main()
        Dim n& = Gl(), r& = 1, c% = 0, i&
        For i = 1 To n
            Dim x& = i
            Do While x Mod 5 = 0
                x \= 5
                c -= 1
            Loop
            While x Mod 2 = 0
                x \= 2
                c += 1
            End While
            r = (r * x) Mod 1000000000
        Next i
        For i = 1 To c
            r = (r * 2) Mod 1000000000
        Next i
        Console.WriteLine(r)
    End Sub
    
    Shared Function Gs$(): Gs = Console.ReadLine(): End Function
    Shared Function Gi%(): Gi = Int32.Parse(Gs()):  End Function
    Shared Function Gl&(): Gl = Int64.Parse(Gs()):  End Function
End Class
